### ISSUE TITLE
Dapper performance compared to ADO

### ISSUE THREAD

**Issue by horseznguyen** [StackExchange/Dapper#1719]
Hi,

I call stored procedure in the following ways:

**Dapper:**

![image](https://user-images.githubusercontent.com/28267489/137838586-66d31a56-0c2f-4fe7-b1c8-7439521da839.png)

**ADO:**

![image](https://user-images.githubusercontent.com/28267489/137838814-3c054969-62e3-42f1-9eb8-44552f71933a.png)

**My request and response ViewModel:**
![image](https://user-images.githubusercontent.com/28267489/137838924-3cb00949-647e-4af8-83f7-d1cd6c547dca.png)
![image](https://user-images.githubusercontent.com/28267489/137838955-5fe77735-1158-4d50-8998-3d58a33b9261.png)

**My params of stored procedure:**

![image](https://user-images.githubusercontent.com/28267489/137842582-56c4f71b-70fa-4bb6-b12a-5bf47e345c14.png)

**Host environment** : IIS

So the problem is I use Apache JMeter to test the performance, the result is as follows:
![image](https://user-images.githubusercontent.com/28267489/137842269-11f494ae-62af-46d2-bfa4-fe0f3aca0b4e.png)

i don't know if i misconfigured dapper, but the time difference is quite big, and it crashes the whole website when i use dapper, ADO doesn't

Thanks.

**Comment by arnederuwe** [comment]
Any reason why you're not awaiting `QueryMultipleAsync`?
Like this:
```
using (var multiple = await connection.QueryMultipleAsync(...))
{
     var x = await multiple.ReadSingleAsync<int>();
     var y = await multiple.ReadAsync<...>();
}
```

**Comment by mgravell** [comment]
@arnederuwe  the code appears to use the new inline `using` syntax; should be fine

**Comment by mgravell** [comment]
@horseznguyen indeed, the number of times you should use `.Wait()` or `.Result` (or `.GetAwaiter().GetResult()`) is *basically* zero (†); `await` should always be preferred


†=there is a caveat around things that are known to have already completed, ideally successfully, and in particular when optimizing a tight loop with fast-path/slow-path duality, but: that doesn't apply here
